### PR TITLE
test: Add MC/DC tests for loop pattern detector (stuck_detector)

### LINKED ISSUE #11599: Improve test coverage for _is_stuck_action_observation_pattern with MC/DC
**What problem or use case are you trying to solve?**

The _is_stuck_action_observation_pattern method in stuck.py is critical for loop detection. It contains complex boolean logic (e.g., in the actions_equal variable) that is not fully covered by the existing test suite. The current test (test_is_stuck_repeating_action_observation_pattern) only validates the "happy path" where a loop is successfully detected.

This lack of coverage means we cannot be fully confident that the detector won't miss a loop (false negative) or incorrectly report a loop (false positive) in various edge cases.

**Describe the UX or technical implementation you have in mind**

I propose enhancing the unit tests for this method in tests/unit/controller/test_stuck.py by applying the Modified Condition/Decision Coverage (MC/DC) methodology.

The technical implementation will involve:

    - Analyzing the key decisions, specifically D1 (the length check) and D2 (the actions_equal logic).

    - Creating the necessary N+1 test cases required to prove the independence of each logical condition.

    - Implementing these new tests to validate all the failure scenarios (e.g., where the pattern is broken in different ways), not just the success scenario.

**Additional context**

This change will significantly increase the robustness of the StuckDetector and give us higher confidence in its behavior. I am currently working on this analysis and implementation and plan to submit a PR soon.

### PR DESCRIPTION
**Pull request by phenric26**
## Summary of PR

This PR enhances the robustness and test coverage of the StuckDetector by adding 6 new unit test cases for the _is_stuck_action_observation_pattern method.

Why this is needed: The existing test for this method only covered the "happy path" (the A-B-A-B pattern being successfully detected). The complex boolean logic in the D1 (len_check) and D2 (actions_equal) decisions was not being tested for independent failures.

These new tests implement Modified Condition/Decision Coverage (MC/DC), ensuring that each condition (CD1 through CD6) is shown to independently affect the decision's outcome.

Summary of Changes:

    Adds 6 new tests to tests/unit/controller/test_stuck.py.

    The tests validate the 3 MC/DC cases for the D1 Decision (length check).

    The tests validate the 5 MC/DC cases for the D2 Decision (actions_equal), covering all pattern-breaking failures (e.g., test_fail_pattern_break_A6_A4).

    Adds reusable event Mocks (Ap, Ai, Adp, etc.) to the top of the test file for readability.

    Adds a lightweight fixture (stuck_detector_mcdc) to isolate these tests.

Result: All tests (new and old) are passing. The test count in the test_stuck.py file has increased from 22 to 28, confirming the robustness of the loop detection logic.

Screenshot from 2025-10-31 01-19-50

## Change Type

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Refactor
- [X] Other (dependency update, docs, typo fixes, etc.)

## Checklist

- [X] I have read and reviewed the code and I understand what the code is doing.
- [X] I have tested the code to the best of my ability and ensured it works as expected.

## Fixes

Resolves #11599

### CODE REVIEW

File: tests/unit/controller/test_is_stuck.py [hunk 1]
```diff
@@ -793,6 +825,149 @@ def test_is_not_stuck_context_window_error_in_non_headless(self, stuck_detector)
...
+            Ap, Op, E_outro, Ai, Oi,  # A4, O4, A5, O5
+            Ap, Op, E_outro           # A6, O6
+        ]
+
+        assert stuck_detector_mcdc._is_stuck_action_observation_pattern(history, 0) is True
+
+    def test_loop_ignores_extra_actions(self, stuck_detector_mcdc: StuckDetector):
+
+        history: list[Event] = [
+            Ap, # Ação extra (A0), será ignorada
```
Comment (enyst) [inline tests/unit/controller/test_is_stuck.py]: Sorry @phenric26  , could we please use English? 😄

File: .gitignore [hunk 2]
```diff
@@ -259,3 +259,5 @@ containers/runtime/code
 test-results
 .sessions
 .eval_sessions
+Miniforge3-Linux-x86_64.sh
```
Comment (enyst) [inline .gitignore]:
```suggestion
```
Nit: probably should not ignore this?

**Comment by mamoodi** [comment]
@phenric26 any interest in addressing the comments left by enyst?

**Comment by enyst** [comment]
@openhands Understand the goal of this PR.

1. set up the project with make build, then run pre-commit on this PR and fix the issues. Note that the PR is from a fork branch phenric26:feature/mcdc-stuck-detector-tests , so handle the remote correctly.
2. Fix the tests inline with existing project patterns like top-level imports, not inline, as much as possible, and check and dedup these tests with the existing tests on stuck detector (find them)

**Comment by neubig** [comment]
Hey @enyst , this has been open for a while, if you're reviewing could you take another look?

File: .gitignore [hunk 3]
```diff
@@ -262,3 +262,4 @@ containers/runtime/code
 test-results
 .sessions
 .eval_sessions
+venv/
```
Comment (enyst) [inline .gitignore]:
```suggestion
```
